api/konnect/v1alpha1/konnect_cloudgateway_transitgateway_types.go

```diff
@@ -0,0 +1,199 @@
+// +kubebuilder:subresource:status
+// +kubebuilder:subresource:finalizer
+// +apireference:kgo:include
+// +kubebuilder:printcolumn:name="State",type=string,JSONPath=`.status.state`,description="State of the transit gateway in Konnect"
+// +kong:channels=gateway-operator
+type KonnectCloudGatewayTransitGateway struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of KonnectCloudGatewayTransitGateway.
+	Spec KonnectCloudGatewayTransitGatewaySpec `json:"spec"`
+
+	// Status defines the observed state of KonnectCloudGatewayTransitGateway.
+	Status KonnectCloudGatewayTransitGatewayStatus `json:"status,omitempty"`
+}
+
+// KonnectCloudGatewayTransitGatewaySpec defines the desired state of KonnectCloudGatewayTransitGateway.
+//
+// +kubebuilder:validation:XValidation:rule="self.networkRef.type == 'namespacedRef'", message = "only namespacedRef is supported currently"
+type KonnectCloudGatewayTransitGatewaySpec struct {
+	// NetworkRef is the schema for the NetworkRef type.
+	//
+	// +kubebuilder:validation:Required
+	NetworkRef commonv1alpha1.ObjectRef `json:"networkRef"`
+	// KonnectTransitGatewayAPISpec is the configuration of the transit gateway on Konnect side.
+	KonnectTransitGatewayAPISpec `json:",inline"`
+}
+
+// TransitGatewayType defines the type of Konnect transit gateway.
+type TransitGatewayType string
+
+const (
+	// TransitGatewayTypeAWSTransitGateway defines the the AWS transit gateway type.
+	TransitGatewayTypeAWSTransitGateway TransitGatewayType = "awsTransitGateway"
+	// TransitGatewayTypeAzureTransitGateway defines the Azure transit gateway type.
+	TransitGatewayTypeAzureTransitGateway TransitGatewayType = "azureTransitGateway"
+)
+
+// KonnectTransitGatewayAPISpec specifies a transit gateway on the Konnect side.
+// The type and all the types it referenced are mostly copied github.com/Kong/sdk-konnect-go/models/components.CreateTransitGatewayRequest.
+//
+// +kubebuilder:validation:XValidation:rule="self.type == oldSelf.type", message="spec.type is immutable"
+// +kubebuilder:validation:XValidation:rule="self.type == 'awsTransitGateway' ? has(self.awsTransitGateway) : true", message="must set spec.awsTransitGateway when spec.type is 'awsTransitGateway'"
+// +kubebuilder:validation:XValidation:rule="self.type != 'awsTransitGateway' ? !has(self.awsTransitGateway) : true", message="must not set spec.awsTransitGateway when spec.type is not 'awsTransitGateway'"
+// +kubebuilder:validation:XValidation:rule="self.type == 'azureTransitGateway' ? has(self.azureTransitGateway) : true", message = "must set spec.azureTransitGateway when spec.type is 'azureTransitGateway'"
+// +kubebuilder:validation:XValidation:rule="self.type != 'azureTransitGateway' ? !has(self.azureTransitGateway) : true", message = "must not set spec.azureTransitGateway when spec.type is not 'azureTransitGateway'"
+// TODO: add more constraints on attachment_config based on type.
```

Comment:
Is this done already perhaps? (I can see the validation rules above)

Reply:
done.